syft/execution/role.py

```diff
@@ -133,7 +132,18 @@ def _execute_action(self, action):
         return_placeholder = self._fetch_placeholders_from_ids(return_placeholder)
 
         if _self is None:
-            response = eval(cmd)(*args, **kwargs)  # nosec
+            cmd_path = cmd.split(".")
+
+            package_name = cmd_path[0]
+            subpackage_names = cmd_path[1:-1]
+            method_name = cmd_path[-1]
+
+            package = framework_packages[package_name]
+            for subpackage_name in subpackage_names:
+                package = getattr(package, subpackage_name)
+            method = getattr(package, method_name)
```

Comment:
Do you think we could put this in a `execute_command` method to make it easier to read?

Reply:
Yeah, an `Extract Method` refactor is a good idea here. 👍